tests/ci/cdk/cdk/ssm/bm_framework_ssm_document.yaml

```diff
@@ -0,0 +1,51 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+---
+schemaVersion: '2.2'
+description: aws-lc:bmFrameworkEc2Benchmark
+mainSteps:
+  - action: aws:runShellScript
+    name: runShellScript
+    inputs:
+      timeoutSeconds: '7200'
+      runCommand:
+        - sudo -i
+        - export DEBIAN_FRONTEND=noninteractive
+        - export CPU_TYPE=$(dpkg --print-architecture)
+        # if we want to disable CPU features, pass in the {OPENSSL_ia32cap} value
+        - if [ {OPENSSL_ia32cap} ]; then export OPENSSL_ia32cap={OPENSSL_ia32cap}; fi
+        - echo "${OPENSSL_ia32cap}"
+        # if we have a cpu type of x86, we want linux-x86
+        - if [ "${CPU_TYPE}" = amd64 ]; then export CPU_ARCH=linux-x86; fi
+        # if we have a cpu type of arm, we want linux-aarch
+        - if [ "${CPU_TYPE}" = arm64 ]; then export CPU_ARCH=linux-aarch; export IMAGE_TAG_POST=-aarch; fi
+        # create bm_framework directory and checkout aws-lc
+        - mkdir bm_framework
+        - cd bm_framework
+        - git clone {GITHUB_REPO} aws-lc-pr
+        - git clone https://github.com/awslabs/aws-lc.git aws-lc-prod
+        - git clone https://boringssl.googlesource.com/boringssl
+        - git clone https://github.com/openssl/openssl.git
+        # install docker
+        - apt-get update
+        - apt-get install -y apt-transport-https ca-certificates curl gnupg lsb-release
+        - curl -fsSL https://download.docker.com/linux/ubuntu/gpg | gpg --dearmor -o /usr/share/keyrings/docker-archive-keyring.gpg
+        # need to change below depending on cpu type
+        - echo "deb [arch="${CPU_TYPE}" signed-by=/usr/share/keyrings/docker-archive-keyring.gpg] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable" | tee /etc/apt/sources.list.d/docker.list > /dev/null
+        - apt-get update
+        - apt-get install -y docker-ce docker-ce-cli containerd.io
+        # build docker image and get dependencies from it
+        - cd aws-lc-pr
+        - git checkout {COMMIT_ID}
+        - cd ../
+        - cd openssl
+        - git checkout OpenSSL_1_1_1-stable
+        - cd ../
```

Comment:
* For openssl, the commit/code in that branch can still get changed. If the code is not locked, it maybe unclear to reproduce some benmark.
* Nit: this can be condensed by `git clone --branch {branch} url`.

Reply:
Do you know of a way to git clone a locked version of the branch? Is there a specific commit that I should be benchmarking against?

Reply:
Answered https://github.com/awslabs/aws-lc/pull/209#discussion_r683068518